#VRML_SIM R2022a utf8
WorldInfo {
  info [
    "Example use of a DistanceSensor device."
  ]
  title "Distance Sensor"
}
Viewpoint {
  orientation -0.36749958339737343 0.06940173917454177 0.927430566027617 2.794829858426511
  position 5.0118774013660214 -1.0666724346867054 2.4008082972714044
  follow "MyBot:ds0"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 2.88 0 0
  floorSize 2 2
  floorTileSize 1 1
  wallHeight 0.2
}
RectangleArena {
  translation 2.39 -0.49 -0.23
  name "rectangle arena(1)"
  wallHeight 0.2
}
Robot {
  translation 2.21894 0.498343 0.027094
  rotation 0.014228120641848613 0.042869594246863915 -0.9989793583813983 0
  children [
    Transform {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 115.17351728120123
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.04499999999999727 0.025
        rotation -0.617249138033943 0.617249316259413 -0.48785938873167284 2.2338164168979184
        children [
          DEF WHEEL Transform {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 52.230339752900605
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.0450000000000003 0.025
        rotation -0.6541664744851412 0.6541664185454199 -0.3796478901655319 2.415914132347849
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation -0.9626378392751118 0.19147896803009973 0.19147896803009973 -1.608865
      children [
        DEF SENSOR Transform {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds4"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.02 -0.042 0.063
      rotation -0.7268313515069925 0.4856522348690052 0.4856522348690052 -1.884578
      children [
        USE SENSOR
      ]
      name "ds5"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0 -0.046 0.063
      rotation -0.617255312726128 0.617255312726128 -0.4878439892247724 2.23382
      children [
        USE SENSOR
      ]
      name "ds6"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.0449924 8.32881e-07 0.062199
      rotation 0.9997945828626467 -0.014327294022322573 0.014335994018692734 1.57102
      children [
        USE SENSOR
      ]
      name "ds3"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0 0.046 0.063
      rotation 0.6172512718011021 0.6172512718011021 0.4878542148222601 2.23382
      children [
        USE SENSOR
      ]
      name "ds0"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.02 0.042 0.063
      rotation -0.7268278184260777 -0.4856548786751704 -0.4856548786751704 -1.884582
      children [
        USE SENSOR
      ]
      name "ds1"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation -0.9626362180015103 -0.19148304336382932 -0.19148304336382932 -1.608867
      children [
        USE SENSOR
      ]
      name "ds2"
      lookupTable [
        0 0 0
        0.02 0 0
        2 2 0
      ]
      aperture 1
    }
    DEF SMILE Transform {
      translation 0.036 0 0.025
      rotation 0.25056300542381216 -0.9351130202419243 -0.25056300542381216 1.637834
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.721569 0.290196 0.290196
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.018
            radius 0.009
            subdivision 3
          }
        }
      ]
    }
    Transform {
      rotation 0 0 1 3.14
      children [
        USE SMILE
      ]
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "CheapSLAM"
}
